sam_learning/learners/sam_learning.py

```diff
@@ -222,6 +224,18 @@ def construct_safe_actions(self) -> None:
         """Constructs the single-agent actions that are safe to execute."""
         pass
 
+    def handle_negative_preconditions(self) -> None:
+        """Function that handles ignoring negative preconditions."""
+        if not self.ignore_negative_preconditions:
+            return
+        for a in self.partial_domain.actions.values():
+            new_preconditions = set()
+            for precondition in a.preconditions.root.operands:
+                if isinstance(precondition, Predicate) and not precondition.is_positive:
+                    continue
+                new_preconditions.add(precondition)
+            a.preconditions.root.operands = new_preconditions
```

Comment:
new line (these new lines make the code structure much clearer so be mindful to add them in the correct locations).